### DISCUSSION TITLE
[form] add  form.isChanged() ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [mantinedev/mantine#3519] [Feature requests]
when any field changed, i need change  submit button style

**Comment by noahsb** [comment]
Use `form.isDirty()`

https://mantine.dev/form/status/#touched-and-dirty-state

**Reply by vaynevayne** [reply]
isDirty is compare between form.values with  initinlValues 
isChanged is compare between form.values with  form.lastValues

**Reply by noahsb** [reply]
You should be able to accomplish both comparisons with `isDirty` using it in combination with [resetDirty](https://mantine.dev/form/status/#resettouched-and-resetdirty)

Can you post a code sample? I might be able to help if there's a particular situation you're stuck on

**Reply by vaynevayne** [reply]
```jsx
 const [isChanged, setIsChanged] = useState(false)
  useEffect(() => {
    setIsChanged(true)
  }, [form.values])

  const onSubmit = (values: FormValues) => {
    setIsChanged(false)
  }
  <form onSubmit={form.onSubmit(onSubmit)}>
   <Button type='submit'  variant={isChanged ? 'gradient' : 'filled'}>search</Button>
  </form>
```

like above

**Reply by noahsb** [reply]
If the only time your initial values can become out of sync is on submit, this should be the same:

```
  const onSubmit = (values: FormValues) => {
    form.resetDirty();
  }

  <form onSubmit={form.onSubmit(onSubmit)}>
   <Button type='submit' variant={form.isDirty() ? 'gradient' : 'filled'}>search</Button>
  </form>
```

**Reply by vaynevayne** [reply]
How is it possible that there is only one button in a form? this is just a demo. There is nothing to say.

**Reply by noahsb** [reply]
Sorry, I don’t understand. What are you wanting to change? You can only have one _submit_ button in a form but you can have multiple buttons in a form if desired
